### DISCUSSION TITLE
Using GPUs that lack support for double-precision floating point operations

### DISCUSSION THREAD

**Discussion by PabloPiaggi** [deepmodeling/deepmd-kit#3805] [Q&A] [answered]
Hello! I am looking into buying NVIDIA L40S GPUs, which only have support for single-precision floating point operations. I would like to know if anybody has experience running DeePMD on this type of hardware and what they experience has been. Furthermore, I have been reading the manual to understand the input needed for training and inference (with LAMMPS) using single precision. I understood that the only change needed is setting DP_INTERFACE_PREC=low. Is this correct? Please let me know.

Many thanks!

Pablo

**Comment by njzjz** [comment, marked as answer]
We suggest a mixed precision, where only the precision of NN is FP32 and other parts (environmental matrix, output energies) are still FP64. https://github.com/deepmodeling-activity/deepmd-kit-v2-paper/blob/main/models/04/input.json provides an example.

**Reply by PabloPiaggi** [reply]
Great! Thanks for the suggestion.
